File: java/activity/ViewPagerActivity.java
/*yourpackagename*/



public class ViewPagerActivity extends AppCompatActivity() {

    //set that ViewPager
    private ViewPager viewPager;

   
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_view_pager);
        
     // set fragment on adapter
    private void setupViewPager(ViewPager viewPager){
        View_Pager_Acitivity.ViewPagerAdapter adapter = new View_Pager_Acitivity.ViewPagerAdapter(getSupportFragmentManager());
        adapter.addFrag(new Home(), "tab1");
        adapter.addFrag(new LoginDetail(), "tab2");
        viewPager.setAdapter(adapter);
    }


    //make class for  adapter
    static class ViewPagerAdapter extends FragmentPagerAdapter {
        private final List<Fragment> mFragmentList = new ArrayList<>();
        private final List<String> mFragmentTitleList = new ArrayList<>();

        public ViewPagerAdapter(FragmentManager manager){
            super(manager);
        }

        @Override
        public Fragment getItem(int position) {
            return mFragmentList.get(position);
        }

        @Override
        public int getCount() {
            return mFragmentList.size();
        }

        public void addFrag(Fragment fragment, String title){
            mFragmentList.add(fragment);
            mFragmentTitleList.add(title);
        }

        @Override
        public CharSequence getPageTitle(int position){
            return mFragmentTitleList.get(position);
        }
    }
    }

}

File: kotlin/activity/ViewPagerActivity.kt
/*yourpackagename*/


import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import kotlinx.android.synthetic.main.activity_view_pager.*
import kotlinx.android.synthetic.main.appbar_main.*

class ViewPagerActivity : AppCompatActivity() {

    private var pagerAdapter: CostumViewPagerAdapter?=null

   

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_view_pager)
     
        setSupportActionBar(toolbar_main)

        /* add your view pager on here */

        pagerAdapter = CostumViewPagerAdapter(supportFragmentManager)
        pagerAdapter!!.addFragments(Tab1Fragment(),"TAB1")
        pagerAdapter!!.addFragments(Tab2Fragment(),"TAB2")

        //adding pageradapter to viewpager
        costomViewPager.adapter = pagerAdapter


        //Now setting up viewpager with tablayout
        costomTabLayout.setupWithViewPager(costomViewPager)
    }
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressed()
            return true
        }
        return false
    }

}
